lib/user_search_string.rb

```diff
@@ -0,0 +1,6 @@
+class UserSearchString < String
+  def initialize(string = '')
+    super string.downcase.gsub /[^#{ ::User::ALLOWED_LOGIN_CHARACTERS }]/, ''
```

Comment:
Ambiguous regexp literal. Parenthesize the method arguments if it's surely a regexp literal, or add a whitespace to the right of the `/` if it should be a division.